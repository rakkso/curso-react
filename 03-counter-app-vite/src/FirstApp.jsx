 
 const miFuncion = () => {
    return "Un literal"
 }
 
 export function FirstApp( prop) {
    return (
    <>
    <h1> FirstApp!!</h1>
    <p> Parrafo {prop.titulo }</p>
    <p>{miFuncion()}</p>
    </>
    );
}