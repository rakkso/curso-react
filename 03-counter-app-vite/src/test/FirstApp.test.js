
  /**
 * @jest-environment jsdom
 */

import { render,screen } from '@testing-library/react'
import { FirstApp } from '../FirstApp'


describe('Pruebas en <FistApp />',() =>{
    test('debe de hacer match con el snapshot',() =>{

        const titulo = 'Hola, Soy Goku'
        const {container } = render (<FirstApp titulo={titulo}/>);

        //console.log(container);
        expect ( container ).toMatchSnapshot();
    })

    test('debe mostrar el título en un h1',() =>{

        const titulo = 'Hola, Soy Goku'
        const {container, getByText, getByTestId } = render (<FirstApp titulo={titulo}/>);
        //console.log(getByText(titulo));
        //miramos que nuestro título este:
        expect (getByText(titulo)).toBeTruthy();
        expect (screen.getByText(titulo)).toBeTruthy();

       
        //miramos que el título esté en un h1, el primero
        const h1 = container.querySelector('h1');
        expect(h1.innerHTML).toBe( titulo);

        expect( getByTestId('test_1').innerHTML).toContain(titulo);

    })
})

