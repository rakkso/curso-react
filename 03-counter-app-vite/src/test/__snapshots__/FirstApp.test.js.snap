// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FistApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test_1"
  >
    Hola, Soy Goku
  </h1>
  <p>
     Parrafo 
    sin subtitulo
  </p>
  <p>
    Un literal
     
    sin valor
  </p>
</div>
`;
